git-multimail/git_multimail.py

```diff
@@ -3227,8 +3307,72 @@ def check_hook_specific_args(options, args):
         return options, args, {'stash_user': options.stash_user,
                                'stash_repo': options.stash_repo}
 
-    # No special options in use, just return what we started with
-    return options, args, {}
+    # Finally, check for gerrit specific arguments
+    required_gerrit_options = (options.oldrev, options.newrev, options.refname,
+                               options.project)
+    if required_gerrit_options == (None,) * 4:
+        # No special options in use, just return what we started with
+        return options, args, {}
+
```

Comment:
@newren Hmm, this considers Gerrit as a kind of special-case, as the code looks like

    if stash is active:
        return stash-options
    if gerrit is not active:
        return the default
    ... gerrit stuff

I'd rather see

    def compute_gerrit_options()
        ...

    if stash is active:
        return stash-options
    if required_gerrit_options != (None,) * 4:
        return compute_gerrit_options()
    return options, args, {}